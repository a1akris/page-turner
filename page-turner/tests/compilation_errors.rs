use page_turner::PageQuery;

#[derive(Debug, Clone, PageQuery)]
struct WithoutKey {
    field1: String,
    field2: String,
    field3: String,
}

#[derive(Debug, Clone, PageQuery)]
struct ZeroStruct;

#[derive(Debug, Clone, PageQuery)]
struct TupleStruct(i32);

#[derive(Debug, Clone, PageQuery)]
enum NotAStruct {
    #[page_key]
    Variant1(String),
    Variant2(String),
}

fn main() {}
