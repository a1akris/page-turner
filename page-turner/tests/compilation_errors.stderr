error: Use #[page_key] attribute to mark a field representing a key
 --> tests/compilation_errors.rs:4:8
  |
4 | struct WithoutKey {
  |        ^^^^^^^^^^

error: Unsupported shape `unit`
  --> tests/compilation_errors.rs:10:24
   |
10 | #[derive(Debug, Clone, PageQuery)]
   |                        ^^^^^^^^^
   |
   = note: this error originates in the derive macro `PageQuery` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `newtype`
  --> tests/compilation_errors.rs:13:24
   |
13 | #[derive(Debug, Clone, PageQuery)]
   |                        ^^^^^^^^^
   |
   = note: this error originates in the derive macro `PageQuery` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `enum`
  --> tests/compilation_errors.rs:16:24
   |
16 | #[derive(Debug, Clone, PageQuery)]
   |                        ^^^^^^^^^
   |
   = note: this error originates in the derive macro `PageQuery` (in Nightly builds, run with -Z macro-backtrace for more info)
